core/domain/exp_services_test.py

```diff
@@ -323,6 +323,56 @@ def test_retrieval_of_explorations(self):
         with self.assertRaises(Exception):
             exp_services.get_exploration_by_id('fake_exploration')
 
+    def test_retrieval_of_multiple_exploration_versions(self):
```

Comment:
Break this into separate tests: one for fake exp ID, one for correct behavior of versions, one for version number errors.

Reply:
Done.